types/agent.d.ts

```diff
@@ -22,6 +23,8 @@ declare namespace Agent {
     maxRedirections?: number;
 
     interceptors?: { Agent?: readonly Dispatcher.DispatchInterceptor[] } & Pool.Options["interceptors"]
+
+    dnsResolverOptions: DNSResolver.Options;
```

Comment:
Similarly, what about having just `dnsResolver` here?

Reply:
It was asked before to rename the options to `dnsResolverOptions` and have a separate one called `dnsResolver` to specify a custom DNSResolver class.

I changed the disable check on `dnsResolver` instead of putting it in the options.

So to disable it will be:

```js
new Agent({
  dnsResolver: false,
})
```